### DISCUSSION TITLE
Sensor with "-1" "-2" in Name (INA226-1) evaluate with script

### DISCUSSION THREAD

**Discussion by andy75andreas** [arendst/Tasmota#18387] [Q&A] [answered]
Hello,

I have connected an INA226 sensor to a solar control unit.
The values are also displayed well in the web interface.
Now I wanted to programme a few automatic processes with a script.
Unfortunately, the values from the INA226 are not taken over in the script.
However, if I take the values from the PZEM, it works.
What am I doing wrong here?
![INA226](https://user-images.githubusercontent.com/47005753/231107295-80625795-5b3a-4b6c-ac05-eae63564993b.JPG)

My script:
`
>D
battv=0
batti=0
cospi0=0
verbrauch=0

>T
battv=INA226-1#Voltage
batti=INA226-1#Current
cospi0=ENERGY#Factor[0]
verbrauch=ENERGY#Today

>S
print Akku %battv%V
print Akku %batti%A
print Phasenwinkel L1 %cospi0%
print Verbrauch %verbrauch%kWh
`

And the output in the console:
`
09:52:07.312 MQT: tele/Solarsteuerung_8D3EA8/SENSOR = {"Time":"2023-04-11T09:52:07","ENERGY":{"TotalStartTime":"2023-01-31T20:29:01","Total":452.697,"Yesterday":4.439,"Today":1.688,"Period":[0,1,0],"Power":[98,50,9],"ApparentPower":[152,100,80],"ReactivePower":[116,87,80],"Factor":[0.65,0.50,0.11],"Frequency":[50,50,50],"Voltage":[237.20,235.80,237.20],"Current":[0.642,0.426,0.338]},"INA226-1":{"Id":0,"Voltage":13.44,"Current":2.859,"Power":38}}
09:52:07.327 Akku 0.00V
09:52:07.329 Akku 0.00A
09:52:07.331 Phasenwinkel L1 0.65
09:52:07.333 Verbrauch 1.69kWh
`

print Akku %battv%V is always 0.00
print Akku %batti%A is always 0.00

why?

**Comment by sfromis** [comment, marked as answer]
To have any chance of making it work, you need `SetOption64` to use `_` as sensor name delimiter instead of `-`, due to the latter char being seen as a subtraction.

**Reply by andy75andreas** [reply]
Thank you.

now I have all the ads for further use.

**_SetOption34 1_**

`12:20:31.842 MQT: tele/Solarsteuerung_8D3EA8/SENSOR = {"Time":"2023-04-11T12:20:31","ENERGY":{"TotalStartTime":"2023-01-31T20:29:01","Total":453.750,"Yesterday":4.439,"Today":2.741,"Period":[3,1,1],"Power":[252,72,66],"ApparentPower":[348,146,89],"ReactivePower":[240,127,61],"Factor":[0.72,0.50,0.73],"Frequency":[50,50,50],"Voltage":[242.00,244.30,238.80],"Current":[1.439,0.598,0.374]},"INA226_1":{"Id":0,"Voltage":14.11,"Current":10.138,"Power":143}}
12:20:31.857 Akku 14.11V
12:20:31.860 Akku 10.14A
12:20:31.861 Phasenwinkel L1 0.72
12:20:31.863 Verbrauch 2.74kWh`
